### DISCUSSION TITLE
Kor support async functions.

### DISCUSSION THREAD

**Discussion by shividwivedi** [eyurtsev/kor#314] [Q&A]
I am developing document parsing using Langchain kor!, so i want to implement using async functions so that i do openai calling simultaneously. So Kor supports async?

**Comment by eyurtsev** [comment]
Yes. It's using langchain chat models which support async.

However, if your'e already using openai which has native function calling. You will probably get better performance from not using kor, but instead using just the open ai chat model in a function calling mode with reference examples.

https://python.langchain.com/v0.2/docs/concepts/#functiontool-calling

**Reply by shividwivedi** [reply]
okay.
